tests/functional/apps/eos.py

```diff
@@ -0,0 +1,216 @@
+        # response = dataAllowed (1) ||
+        #            LEDGER_MAJOR_VERSION (1) ||
+        #            LEDGER_MINOR_VERSION (1) ||
+        #            LEDGER_PATCH_VERSION (1)
+        assert len(response) == 4
+
+        data_allowed = int(response[0]) == 1
+        major = int(response[1])
+        minor = int(response[2])
+        patch = int(response[3])
+        return data_allowed, (major, minor, patch)
+
+    def send_get_app_configuration(self):
+        return self._client.exchange(CLA, INS.INS_GET_APP_CONFIGURATION, 0, 0, b"")
+
+    def compute_adress_from_public_key(self, public_key):
+
+        head = 0x03 if (public_key[64] & 0x01) == 0x01 else 0x02
+        public_key_compressed = bytearray([head]) + public_key[1:33]
+
+        ripemd = hashlib.new('ripemd160')
+        ripemd.update(public_key_compressed)
+        check = ripemd.digest()[:4]
+
+        buff = b58encode(public_key_compressed + check).decode("ascii")
+        return "EOS" + buff
+
+    def parse_get_public_key_response(self, response: bytes, request_chaincode) -> (bytes, str, bytes):
+        # response = public_key_len (1) ||
+        #            public_key (var) ||
+        #            address_len (1) ||
+        #            address (var) ||
+        #            chain_code (32)
+        offset: int = 0
+
+        public_key_len: int = response[offset]
+        offset += 1
+        public_key: bytes = response[offset:offset + public_key_len]
+        offset += public_key_len
+        address_len: int = response[offset]
+        offset += 1
+        address: str = response[offset:offset + address_len].decode("ascii")
+        offset += address_len
+        if request_chaincode:
+            chaincode: bytes = response[offset:offset + 32]
+            offset += 32
+        else:
+            chaincode = None
+
+        assert len(response) == offset
+        assert len(public_key) == 65
+        assert self.compute_adress_from_public_key(public_key) == address
+
+        return public_key, address, chaincode
+
+    def _get_public_key_params(self, derivation_path: bytes, confirm: bool, request_chaincode: bool):
+        p1 = P1_CONFIRM if confirm else P1_NON_CONFIRM
+        p2 = P2_CHAINCODE if request_chaincode else P2_NO_CHAINCODE
+        return CLA, INS.INS_GET_PUBLIC_KEY, p1, p2, derivation_path
+
+    def send_get_public_key_non_confirm(self, derivation_path: bytes, request_chaincode: bool) -> RAPDU:
+        apdu_params = self._get_public_key_params(derivation_path, False, request_chaincode)
+        return self._client.exchange(*apdu_params)
+
+    @contextmanager
+    def send_async_get_public_key_confirm(self, derivation_path: bytes, request_chaincode: bool):
+        apdu_params = self._get_public_key_params(derivation_path, True, request_chaincode)
+        with self._client.exchange_async(*apdu_params):
+            yield
+
+    def split_message(self, message: bytes) -> List[bytes]:
```

Comment:
This could go in the ragger utils what do you think ?

Reply:
We could!
That's not a real big thing, but still it will be used in many apps probably.

Reply:
If you don't want to tackle this now we can open the issue in ragger

Reply:
done :)